### ISSUE TITLE
Todos reducer is impure

### ISSUE THREAD

**Issue by timdeschryver** [tomastrajan/angular-ngrx-material-starter#294]
The todos reducer is impure at the moment because an `id` is generated when a todo is added.

```ts
case TodosActionTypes.ADD:
  return {
    ...state,
    items: [
      {
        id: uuid(), // this line here
        name: action.payload.name,
        done: false
      }
    ].concat(state.items)
  };
```

This is against the redux principles, I would suggest to generate the `id`  within the action creator `ActionTodosAdd`.

**Comment by tomastrajan** [comment]
Hey @timdeschryver great catch thank you! Would you be also interested in submitting a PR?

**Comment by timdeschryver** [comment]
@tomastrajan I'll let this open and see if someone picks it up first 😄

**Comment by tomastrajan** [comment]
Fixed in #295
